### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by cjalmeida**
This PR removes unneeded runtime dependencies (six, numpy)

### CODE REVIEW

**Comment by wesselb** [comment]
Hey @cjalmeida! Thank you for your interest in the package. This PR looks good to me. You're right that both `six` and `numpy` aren't used, so can be removed as dependencies. The PR, however, also removes the compiled component, which is critical for minimising the overhead that plum brings. Shall we keep that in? I've made a suggestion.

File: setup.py [hunk 1]
```diff
@@ -1,11 +1,9 @@
 from setuptools import find_packages, setup, Extension
 
-requirements = ["numpy>=1.16", "six"]
 
 setup(
     packages=find_packages(exclude=["docs"]),
     python_requires=">=3.6",
-    install_requires=requirements,
-    ext_modules=[Extension("plum.function", ["plum/function.py"])],
+    install_requires=[],
```
Comment (wesselb) [inline setup.py:7]:
```suggestion
    ext_modules=[Extension("plum.function", ["plum/function.py"])],
```

**Comment by gc-ss** [comment]
I would be interested in getting this branch merged in.

Could we add numpy as a dev dependency for test and benchmark in the poetry file?

```
[tool.poetry.dev-dependencies]
numpy = "^1.20.2"
```

**Comment by jph00** [comment]
@wesselb we're thinking about using your cool-looking lib for fastcore. But we can't add a dep on numpy. Any chance you could merge this PR or even just make the edit yourself? Or would you like me to create a new clean PR?

**Comment by wesselb** [comment]
@gc-ss, sorry for not having acted on this timely. I completely forgot about this.

@jph00, certainly! 

I've released a new version, `1.1.0`, which drops the dependencies on `numpy` and `six`.

**Comment by jph00** [comment]
Thanks a lot!

**Comment by wesselb** [comment]
No problem at all!

**Comment by gc-ss** [comment]
> I've released a new version, `1.1.0`, which drops the dependencies on `numpy` and `six`.

Appreciate it Wessel!

**Pull request closed without merging**